docs/signature.md

```diff
@@ -0,0 +1,93 @@
+# Image Signature Specification
+
+**Version 0.1**
+
+## Introduction
+
+This document defines a detached container image signature object and signing methods.
+
+## Signature Format
+
+```js
+{
+    "critical": {/* required fields */
+        "identity": {/* identity reference */},
+        "image": {/* signed object reference */ },
+        "type": "..."
+    },
+    "optional": {/* optional metadata fields */}
+    }
+}
+```
+
+### Fields
+
+There are two top-level fields, **critical** (required) and **optional** (optional).
+
+#### `critical`
+
+**identity** (string):
+
+```js
+{
+    "docker-reference": imageName
+}
+```
+
+`imageName` per [V2 API](https://docs.docker.com/registry/spec/api/#/overview) Required.
```

Comment:
> It is acceptable to ask the user to do an one-time setup to trust key X for example.com…

I missed this distinction in my previous comment, but if your trust policy is “I trust anything that example.com does so long as the asserted name starts with ‘example.com/’”, you should be able to accomplish that same goal with names like `product5/standalone:1.3` and an API like:

    $ tool pull --namer example.com product5/standalone:1.3

which is “go to wherever you usually go for this sort of thing, fetch me `product5/standalone:1.3` and ensure it's been named by example.com”.  Where you get it from isn't important.  If folks want some syntactic sugar around transport → namer lookup, you could map:

    $ tool pull example.com/product5/standalone:1.3

to:

    $ tool pull --namer example.com --engine https://example.com product5/standalone:1.3